src/Microsoft.AspNetCore.WebHooks.Receivers/Filters/WebHookVerifyBodyTypeFilter.cs

```diff
@@ -97,118 +128,153 @@ public void OnResourceExecuting(ResourceExecutingContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            var routeData = context.RouteData;
+            if (!routeData.TryGetWebHookReceiverName(out var receiverName))
+            {
+                return;
+            }
+
+            var bodyTypeMetadata = _bodyTypeMetadata;
+            if (_allBodyTypeMetadata != null)
+            {
+                var receiverBodyTypeMetadata = _allBodyTypeMetadata
+                    .FirstOrDefault(metadata => metadata.IsApplicable(receiverName));
+                if (receiverBodyTypeMetadata == null)
+                {
+                    // WebHookMetadataProvider requires an IWebHookBodyTypeMetadataService implementation for all
+                    // receivers. But, this case may occur if user has no receiver-specific actions i.e. uses only the
+                    // GeneralWebHookAttribute.
+                    _logger.LogCritical(
```

Comment:
Doesn't really seem *critical* to me... https://github.com/aspnet/Logging/wiki/Guidelines#critical

Reply:
About to `throw` an `Exception` indicating the configuration is messed up. Nothing is wrong with the request but the application has a routable action that can't handle any any requests for this receiver.

Reply:
Yeah that isn't critical, it's just an error.

Reply:
:ok: